src/SFML/Audio/SoundFileReaderWav.cpp

```diff
@@ -272,12 +286,51 @@ bool SoundFileReaderWav::parseHeader(Info& info)
             }
             m_bytesPerSample = bitsPerSample / 8;
 
-            // Skip potential extra information (should not exist for PCM)
-            if (subChunkSize > 16)
+            if (format == WaveFormatExtensible)
             {
-                if (m_stream->seek(m_stream->tell() + subChunkSize - 16) == -1)
+                // Extension size
+                Uint16 extensionSize = 0;
+                if (!decode(*m_stream, extensionSize))
+                    return false;
+
+                // Valid bits per sample
+                Uint16 validBitsPerSample = 0;
+                if (!decode(*m_stream, validBitsPerSample))
+                    return false;
+
+                // Channel mask
+                Uint32 channelMask = 0;
+                if (!decode(*m_stream, channelMask))
+                    return false;
+
+                // Subformat
+                char subformat[16];
+                if (m_stream->read(subformat, sizeof(subformat)) != sizeof(subformat))
                     return false;
+
+                if (memcmp(subformat, waveSubformatPcm, sizeof(subformat)) != 0)
+                {
+                    err() << "Unsupported format: extensible format with non-PCM subformat" << std::endl;
+                    return false;
+                }
+
+                if (validBitsPerSample != bitsPerSample)
+                {
+                    err() << "Unsupported format: sample size (" << validBitsPerSample << " bits) and "
+                            "sample container size (" << bitsPerSample << " bits) differ" << std::endl;
+                    return false;
+                }
             }
+
+            // Skip potential extra information
+            Int64 subChunkEnd = subChunkStart + subChunkSize;
+
+            Int64 position = m_stream->tell();
+            if ((position == -1) || (position > subChunkEnd))
```

Comment:
Are these checks really needed? I'd say just jump to the end of the current chunk and go on (that's what was done before, anyway)

Reply:
It would only serve to catch malformed headers. Since the rest of the code doesn't do much of this kind of validation, I'm okay with dropping it.